cfme/__init__.py

```diff
@@ -31,3 +34,51 @@ def __getattribute__(self, attr):
                 return object.__getattribute__(self, 'token')
         else:
             return object.__getattribute__(self, attr)
+
+
+class BaseLoggedInPage(View):
+    """This page should be subclassed by any page that models any other page that is available as
+    logged in.
+    """
+    flash = FlashMessages('div#flash_text_div')
+    help = NavDropdown('.//li[./a[@id="dropdownMenu1"]]')
+    settings = NavDropdown('.//li[./a[@id="dropdownMenu2"]]')
+    navigation = VerticalNavigation('#maintab')
+
+    @property
+    def is_displayed(self):
+        return self.logged_in_as_current_user
+
+    def logged_in_as_user(self, user):
+        if self.logged_out:
+            return False
+
+        return user.name == self.current_fullname
+
+    @property
+    def logged_in_as_current_user(self):
+        return self.logged_in_as_user(self.extra.appliance.user)
```

Comment:
`extra` can contain multiple things, that is the "extra context" of the environment of the tests